shotover-proxy/src/transforms/redis/cache.rs

```diff
@@ -1,457 +1,555 @@
 use crate::error::ChainResponse;
-use crate::frame::{CassandraFrame, CassandraOperation, CassandraResult, Frame, RedisFrame};
-use crate::message::{Message, MessageValue, Messages, QueryType};
+use crate::frame::{CassandraFrame, CassandraOperation, Frame, RedisFrame};
+use crate::message::{Message, Messages};
 use crate::transforms::chain::TransformChain;
 use crate::transforms::{
     build_chain_from_config, Transform, Transforms, TransformsConfig, Wrapper,
 };
-use anyhow::{anyhow, bail, Result};
+use anyhow::{bail, Result};
 use async_trait::async_trait;
-use bytes::{BufMut, Bytes, BytesMut};
-use cassandra_protocol::frame::Version;
+use bytes::Bytes;
+use cassandra_protocol::compression::Compression;
+use cql3_parser::cassandra_statement::CassandraStatement;
+use cql3_parser::common::{FQName, Identifier, Operand, RelationElement, RelationOperator};
+use cql3_parser::select::Select;
 use itertools::Itertools;
+use metrics::{register_counter, Counter};
 use serde::Deserialize;
-use sqlparser::ast::{Assignment, BinaryOperator, Expr, Ident, Query, SetExpr, Statement, Value};
-use std::borrow::Borrow;
-use std::collections::HashMap;
-
-const TRUE: [u8; 1] = [0x1];
-const FALSE: [u8; 1] = [0x0];
+use std::collections::{BTreeMap, HashMap};
+use tracing_log::log::{error, warn};
+
+/// Data is stored in Redis as a Hash (hset/hget) and constructed from the cassandra SELECT statement
```

Comment:
I'm not reviewing the cache changes for this PR, but I understand they are included as part of the CQL rewrite